packages/edit-site/src/components/template-part-converter/convert-to-sticky-group.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * WordPress dependencies
+ */
+import { switchToBlockType } from '@wordpress/blocks';
+import { MenuItem } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	store as blockEditorStore,
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../private-apis';
+
+const {
+	hasStickyPositionSupport,
+	useConvertToGroupButtonProps,
+	useIsPositionDisabled,
+} = unlock( blockEditorPrivateApis );
+
+export default function ConvertToStickyGroup( { selectedClientIds, onClose } ) {
+	const { replaceBlocks } = useDispatch( blockEditorStore );
+	const { clientIds, isGroupable, blocksSelection, groupingBlockName } =
+		useConvertToGroupButtonProps( selectedClientIds );
+
+	const { canRemove, hasParents } = useSelect(
+		( select ) => {
+			const { getBlockParents, canRemoveBlocks } =
+				select( blockEditorStore );
+			return {
+				canRemove: canRemoveBlocks( clientIds ),
+				hasParents: !! getBlockParents( clientIds[ 0 ] ).length,
+			};
+		},
+		[ clientIds ]
+	);
+
+	const isPositionDisabled = useIsPositionDisabled( {
+		name: groupingBlockName,
+	} );
+	const isStickySupported = hasStickyPositionSupport( groupingBlockName );
+
+	const onConvertToGroup = () => {
+		const newBlocks = switchToBlockType(
+			blocksSelection,
+			groupingBlockName
+		);
+
+		if ( newBlocks && newBlocks.length > 0 ) {
+			// Because the block is not in the store yet we can't use
+			// updateBlockAttributes so need to manually update attributes.
+			newBlocks[ 0 ].attributes.layout = {
+				type: 'default',
+			};
+			newBlocks[ 0 ].attributes.style = {
+				...( newBlocks[ 0 ].attributes.style || {} ),
+				position: {
+					type: 'sticky',
+					top: '0px',
+				},
+			};
+			replaceBlocks( clientIds, newBlocks );
+		}
+	};
+
+	// For the button to be visible, the following conditions must be met:
+	// - The block is groupable.
+	// - The block can be removed.
+	// - A grouping block is available.
+	// - The block and theme both support sticky position.
+	// - The block has no parents, so is at the root of the template.
+	if (
+		! isGroupable ||
+		! canRemove ||
+		! groupingBlockName ||
```

Comment:
Til about the concept of a "grouping block"! Does this mean blocks could be grouped with a different container than a Group block?

Reply:
Theoretically yes, but in practice the `groupingBlockName` is set up as the Group block when the core blocks are registered [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/index.js#L266). My understanding is that it's good when dealing with grouping (verb) behaviour to use this abstraction so that (in theory) custom editors that don't use core blocks can still work properly. In practice, I'm not sure how frequently that use case is realised, but for this PR, I thought I'd try to be consistent with the existing grouping logic!